src/vsr/superblock_free_set.zig

```diff
@@ -364,7 +457,8 @@ pub const FreeSet = struct {
 
     /// Decodes the compressed bitset in `source` into `set`.
     /// Panics if the `source` encoding is invalid.
-    pub fn decode(set: *FreeSet, source: []align(@alignOf(usize)) const u8) void {
+    pub fn decode(set: *FreeSet, source: []align(@alignOf(Word)) const u8) void {
+        assert(!set.opened);
         // Verify that this FreeSet is entirely unallocated.
         assert(set.index.count() == 0);
         assert(set.blocks.count() == 0);
```

Comment:
I am tempted to do a larger refactor in the follow up and just pass in an `[] Word`, in a case of pushing an `if` up :) 

It seems like the original idea was to pass in the slice of bytes, so that the details of encoding are abstracted away, but we are already leaking alingment, and `[]Word` is just going to be shorter.

Reply:
Tried that, and it looks like `[]align(u8) u8` is actually a better API here: while `[]Word` feels simpler, there are just way to many places to mix up the bytes-vs-words indexing, as _some_ amount of conversions has to happen "in the middle" -- we definitely want to use byte-size in CheckpointState.

I've added % asserts to FreeSetEncoded open&checkpoint, that feels like the right place to assert that